include/wlr/backend/drm.h

```diff
@@ -33,6 +33,21 @@ bool wlr_output_is_drm(struct wlr_output *output);
  */
 uint32_t wlr_drm_connector_get_id(struct wlr_output *output);
 
+/**
+ * Leases a given output to the caller. The output must be from the associated
+ * DRM backend.
+ * Returns -errno on error.
+ */
+int wlr_drm_backend_create_lease(struct wlr_backend *backend,
```

Comment:
This function can only take the `wlr_output`, without the `wlr_backend`, since it's easy to get back the DRM backend from the output internally.

Reply:
I should rename it `wlr_drm_create_lease` then, and validate during the first loop that all of the outputs have the same backend.